src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionInterfaceIsTypeTest.java

```diff
@@ -0,0 +1,94 @@
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.design.InterfaceIsTypeCheck;
+
+public class XpathRegressionInterfaceIsTypeTest extends AbstractXpathTestSupport {
+
+    private final String checkName = InterfaceIsTypeCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testOne() throws Exception {
+        final File fileToProcess = new File(getPath(
+                "SuppressionXpathRegressionInterfaceIsType1.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(InterfaceIsTypeCheck.class);
+
+        final String[] expectedViolation = {
+            "3:1: " + getCheckMessage(InterfaceIsTypeCheck.class,
+                    InterfaceIsTypeCheck.MSG_KEY),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+            "/INTERFACE_DEF[./IDENT["
+                + "@text='SuppressionXpathRegressionInterfaceIsType1']]",
+            "/INTERFACE_DEF[./IDENT["
+                + "@text='SuppressionXpathRegressionInterfaceIsType1']]/MODIFIERS",
+            "/INTERFACE_DEF[./IDENT["
+                + "@text='SuppressionXpathRegressionInterfaceIsType1']]/MODIFIERS/LITERAL_PUBLIC"
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+    }
+
+    @Test
+    public void testTwo() throws Exception {
+        final File fileToProcess = new File(getPath(
+                "SuppressionXpathRegressionInterfaceIsType2.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(InterfaceIsTypeCheck.class);
+
+        moduleConfig.addAttribute("allowMarkerInterfaces", "false");
+
+        final String[] expectedViolation = {
+            "3:1: " + getCheckMessage(InterfaceIsTypeCheck.class,
+                    InterfaceIsTypeCheck.MSG_KEY),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+                "/INTERFACE_DEF[./IDENT["
```

Comment:
Indentation seems incorrect (espessially comparing with the same list from testOne). Please recheck formatting

Reply:
Thanks, done, actually, it is testOne whose format is not good and I have changed it.